run ratbagctl on master through run_ratbagctl

bentiss opened the pull request:
Update ratbagctl ratbagd binding by copying the file from piper (do we have license issues doing that?).

Add a new tool run_ratbagctl which allows developers to spawn ratbagd and use the ratbagd from master insterad of the system wide one.

whot commented:
re license issues: no, the piper ratbagd.py is still MIT licensed, see the header. Unless @Hjdskes says his recent changes are GPL only, but I hope not :)

whot commented on dbus/org.freedesktop.ratbag_devel1.conf.in:
```diff
@@ -0,0 +1,20 @@
+                <allow receive_sender="org.freedesktop.ratbag_devel1_@ratbagd_sha@"/>
+        </policy>
+
+</busconfig>
```
This should have a warning in the form of "FOR DEVELOPMENT ONLY. If you don't know where this file came from, it can be safely removed."

whot commented on meson.build:
```diff
@@ -5,6 +5,14 @@ project('libratbag', 'c', 'cpp',
 libratbag_version = meson.project_version().split('.')
 
+ratbagd_dir = join_paths(meson.current_source_dir(), 'ratbagd')
+r = run_command('/usr/bin/git', 'log --format=format:%h -1 -- @0@'.format(ratbagd_dir).split())
```
two things here: why specify the full path to git? meson approach here would be (I think):

```
git = find_program('git')
run_command(git, ....)
````

The other thing is that the meson doc say that this is run from an unspecified directory, you can't assume that we're in the source directory here. This is something we have to move to a script, using `GIT_DIR=$MESON_SOURCE_ROOT/.git` or some pushd/popd combination.

whot commented on meson.build:
```diff
@@ -440,6 +448,22 @@ executable('ratbagd',
+	   dependencies : deps_ratbagd,
+	   include_directories : include_directories('src'),
+	   install : false,
+	   c_args : '-DRATBAG_DBUS_INTF="ratbag_devel1_@0@"'.format(ratbagd_sha),
```
I would like to buy a few letters so we can spell out INTERFACE. Storage is cheap these days, we can afford a few more characters that we never have to type anyway.

whot commented on meson.build:
```diff
@@ -440,6 +448,22 @@ executable('ratbagd',
+)
+
+config_ratbagd_devel = configuration_data()
+config_ratbagd_devel.set('ratbagd_sha', ratbagd_sha)
```
I'd say rename this to config_ratbagd_sha and then re-use it in both configure_files below

whot commented on ratbagd/ratbagd.h:
```diff
@@ -38,6 +38,10 @@
+#ifndef RATBAG_DBUS_INTF
+#define RATBAG_DBUS_INTF	"ratbag1"
+#endif
+
```
I wonder if it'd be better to just `#define` all the object path and dbus prefixes in ratbagd.h and using them instead of the `RATBAG_DBUS_INTF` sprinkled everywhere

whot commented on tools/run_ratbagctl.in:
```diff
@@ -0,0 +1,51 @@
+# FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
+# DEALINGS IN THE SOFTWARE.
+
+from __future__ import print_function
```
the future is here! this is a python3 thing

whot commented on tools/run_ratbagctl.in:
```diff
@@ -0,0 +1,51 @@
+
+# first copy the policy for the ratbagd daemon to be allowed to run
+shutil.copy(os.path.join(SCRIPT_DIR, 'org.freedesktop.ratbag_devel1.conf'),
+            '/etc/dbus-1/system.d/')
```
if we run this through `configure_file` anyway, why not replace the SCRIPT_DIR with the meson builddir?

whot commented on tools/run_ratbagctl.in:
```diff
@@ -0,0 +1,51 @@
+ratbagctl = subprocess.Popen(" ".join([os.path.join(SCRIPT_DIR, "ratbagctl"), *sys.argv[1:]]),
+                             shell=True,
+                             env={"RATBAGCTL_DEVEL": "org.freedesktop.ratbag_devel1_@ratbagd_sha@"})
+
```
we prob. need to wrap this into a `try: except KeyboardInterrupt` so we shut down nicely on interrupt

whot commented:
fwiw, I pushed the quick fixes for the above to: https://github.com/whot/libratbag/tree/wip/run-ratbagctl
Please squash them and drop the ones you don't like, thanks.

Hjdskes commented:
> Unless @Hjdskes says his recent changes are GPL only, but I hope not :)

MIT is fine. OTOH, you might want to remove the GObject properties and signals, to make it pure Python again. Should be a straightforward search and replace mostly.

bentiss commented:
> OTOH, you might want to remove the GObject properties and signals, to make it pure Python again. Should be a straightforward search and replace mostly.

I would rather keep both file as close as possible. And honestly, we do not care if rabtagctl is Gobject or pure python (I think). I people start complaining, we can always strip these out, but for now, it'll be easier to have both piper and ratbagctl sharing the same code base (close enough at least).

bentiss commented:
OK, I squashed your changes into my branch and rebased on top of master.
I also removed a couple of extra empty strings from automatic replacements (RATBAG_... "").
